<launch>
    <arg name="raw_data" default="/data/kitti/raw_data"/>
    <arg name="date" default="2011_09_26"/>
    <arg name="id" default="19"/>

    <arg name="id4d" default="$(eval str(arg('id')).zfill(4))"/>
    <arg name="sync" value="$(arg date)_drive_$(arg id4d)_sync"/>

    <arg name="calib_dir" value="$(arg raw_data)/$(arg date)_calib/$(arg date)"/>
    <arg name="data_dir" value="$(arg raw_data)/$(arg sync)/$(arg date)/$(arg sync)"/>

    <param name="use_sim_time" value="true" />
	  <node pkg="kitti_to_rosbag" type="kitti_live_node" name="kitti_to_rosbag" 
        args="$(arg calib_dir) $(arg data_dir)" output="log">
    </node>

    <node pkg="scan_filter" type="scan_filter_node" name="scan_filter" output="log">
        <param name="leaf_size" type="double" value="0.5" />
        <remap from="~/raw_scan" to="/velodyne_points"/>
    </node>

    <node pkg="pcl_processing" type="pcl_server.py" name="pcl_alignment_node" output="screen"/>
    <node pkg="pcl_processing" type="test.py" name="pcl_alignment_test_node" output="screen"/>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pcl_processing)/rviz/test_regist.rviz" />

</launch>